extensions/tls-registry/runtime/src/main/java/io/quarkus/tls/runtime/VertxCertificateHolder.java

```diff
@@ -125,6 +132,37 @@ public synchronized SSLOptions getSSLOptions() {
         return options;
     }
 
+    boolean warnIfOldProtocols(Set<String> protocols, String name) {
+        var list = protocols.stream().map(String::toLowerCase).map(String::trim).toList();
+        boolean warned = false;
+
+        // Check for SSL protocols
+        if (list.stream().anyMatch(p -> p.startsWith("ssl"))) {
+            LOGGER.warnf("Insecure SSL protocol is enabled in TLS bucket '%s'." +
+                    " It is strongly recommended to disable SSL protocols (SSLv2, SSLv3), and use at least TLSv1.3.", name);
+            warned = true;
+        }
+
+        // Check for old TLS protocols (TLSv1.0, TLSv1.1)
+        if (list.contains("tlsv1") || list.contains("tlsv1.1")) {
+            LOGGER.warnf("Insecure TLS protocol TLSv1.0 or TLSv1.1 is enabled in TLS bucket '%s'." +
+                    " It is strongly recommended to disable TLSv1.0 and TLSv1.1, and use at least TLSv1.3.", name);
+            warned = true;
+        }
+
+        // Check if TLSv1.3 or higher is available (future-proof for TLSv1.4+)
+        boolean hasModernTls = list.stream().anyMatch(p -> p.matches("tlsv1\\.[3-9]") || p.matches("tlsv1\\.[1-9][0-9]+") ||
+                p.matches("tlsv[2-9].*"));
```

Comment:
Do we really need a regex here? Asking because it would be nice to avoid regex lookups at startup...

Reply:
Sure yeah, but maybe we can do some bespoke parsing instead?

Reply:
Split, indexof and contains should be enough

Reply:
Well, tried, and it failed lamentably - it 80% worked and contained overly complicated code (and has the great idea to modify the tests instead of fixing the implementation).

Anyway, fixed and it worked now.